cypress/support/cypress-commands/commands.js

```diff
@@ -833,3 +834,294 @@ Cypress.Commands.add('clearCache', () => {
+      if (response === CONSTANTS.NO_RESPONSE) {
+        assert(false, CONSTANTS.NO_MATCHED_RESPONSE);
+      }
+
+      response = typeof response === CONSTANTS.TYPE_STRING ? JSON.parse(response) : response;
+
+      if (
+        response &&
+        response.error &&
+        response.error.message &&
+        CONSTANTS.ERROR_LIST.includes(response.error.message)
+      ) {
+        if (UTILS.getEnvVariable(CONSTANTS.CERTIFICATION) == true) {
+          fireLog.assert(false, `${target} does not support method: ${method}`);
+        } else {
+          fireLog.info(`NotSupported: ${target} does not support method: ${method}`).then(() => {
+            throw new Error(CONSTANTS.STEP_IMPLEMENTATION_MISSING);
+          });
+        }
+      }
+
+      if (task == CONSTANTS.TASK.REGISTEREVENT) {
+        if (response && response.result && response.result.hasOwnProperty(CONSTANTS.LISTENING)) {
+          const eventResponse = {
+            eventListenerId: response.result.event + '-' + response.id,
+            eventListenerResponse: response.result,
+          };
+          response.result = eventResponse;
+        }
+        if (response && response.error && response.error.message) {
+          fireLog.assert(
+            false,
+            `Event registration failed for event ${method} with error message: ${response.error.message} `
+          );
+        }
+      }
+
+      cy.updateResponseForFCS(method, params, response).then((updatedResponse) => {
+        // Create a deep copy to avoid reference mutation
+        const dataToBeCensored = _.cloneDeep(response);
+
+        // Call the 'censorData' command to hide sensitive data
+        cy.censorData(method, dataToBeCensored).then((maskedResult) => {
+          const appLog = target === CONSTANTS.PLATFORM ? 'Firebolt platform' : `app: ${appId}`;
+          fireLog.info(`Response from ${appLog}: ${JSON.stringify(maskedResult)}`);
+        });
+        // Creating object with event name, params, and response etc and storing it in a global list for further validation.
+        const apiOrEventAppObject =
+          task === CONSTANTS.TASK.REGISTEREVENT
+            ? new eventObject(method, params, context, updatedResponse, appId, expected)
+            : new apiObject(method, params, context, updatedResponse, expected, appId);
+
+        const globalList =
+          task === CONSTANTS.TASK.REGISTEREVENT
+            ? CONSTANTS.GLOBAL_EVENT_OBJECT_LIST
+            : CONSTANTS.GLOBAL_API_OBJECT_LIST;
+        UTILS.getEnvVariable(globalList).push(apiOrEventAppObject);
+      });
+    } else {
+      fireLog.info(
+        `${target} returned response in invalid format, which could lead to failures in validations. Response must be in JSON RPC format - ${response}`
+      );
+    }
+  });
+});
+
+Cypress.Commands.add('methodorEventResponseValidation', (validationType, additionalParams) => {
+  const { method, context, contentObject, expectingError, appId, eventExpected } = additionalParams;
```

Comment:
the new commands look good. 
methodorEventResponseValidation and sendMessageToPlatformOrApp